packages/libs/error-stack/src/fmt/hook.rs

```diff
@@ -436,22 +444,23 @@ impl<T: 'static> HookContext<'_, T> {
     }
 }
 
-type BoxedHook = Box<dyn for<'a> Fn(&Frame, &mut HookContext<'a, Frame>) -> Emit + Send + Sync>;
-type BoxedFallbackHook =
+type BoxedHook =
     Box<dyn for<'a> Fn(&Frame, &mut HookContext<'a, Frame>) -> Option<Emit> + Send + Sync>;
+type BoxedFallbackHook =
+    Box<dyn for<'a> Fn(&Frame, &mut HookContext<'a, Frame>) -> Vec<Emit> + Send + Sync>;
```

Comment:
Do you think it makes sense to also allow multiple `Emit`s for `BoxedHook`?

Reply:
I tried an initial implementation with a new type, `Trace`. This type ensures that we guarantee that at least a single `Emit` is emitted.

What do you think about this approach? We could use a plain `Vec<Emit>` instead.

I feel like using a new type `Trace` might be more ergonomic:

```rust
Report::install_debug_hook::<u64>(|val, ctx| {
	Trace::next(format!("u64: {val}"))
})
```

vs.

```rust
Report::install_debug_hook::<u64>(|val, ctx| {
	vec![Emit::next(format!("u64: {val}"))]
})
```

I'd expect that the vast majority only uses a single `Emit`. Therefore, I'd like to make that scenario as easy as possible.

`Trace` also ensures that at least a single value is emitted (even if it is just an `Emit::Hidden`, but we can lift that requirement)

(but I could also overengineer things once again)

Reply:
addressed in #976